packages/core/src/session.ts

```diff
@@ -55,7 +55,8 @@ class Session {
   private _writeConnectionHolder: ConnectionHolder
   private _open: boolean
   private _hasTx: boolean
-  private _lastBookmark: Bookmark
+  private _bookmarkIn: Bookmark
+  private _bookmarkOut?: string
```

Comment:
Maybe it could be called `currentBookmark` and `lastReceivedBookmark`, or something like this. Because the `_bookmarkIn` is being changed when a new bookmark arrives.

Reply:
I'm fine with that :ok_hand: Will do.